pandas/tseries/period.py

```diff
@@ -811,6 +811,35 @@ def _format_native_types(self, na_rep=u('NaT'), date_format=None, **kwargs):
         values[imask] = np.array([formatter(dt) for dt in values[imask]])
         return values
 
+    def _fillna_mask(self, mask, value, downcast=None):
+        if isinstance(value, Period):
+            if value.freq != self.freq:
+                msg = _DIFFERENT_FREQ_ERROR.format(self.freqstr, value.freqstr)
+                raise ValueError(msg)
+
```

Comment:
you don't need the raise Here, instead do this in ``putmask``, in fact ``putmask`` should just raise on any invalid conversions